modules/imap/functions.php

```diff
@@ -1144,3 +1144,21 @@ function get_personal_ns($imap) {
     );
 }}
 
+/**
+ * @subpackage imap/functions
+ */
+if (!hm_exists('parse_sieve_config_host')) {
+function parse_sieve_config_host($host) {
+    $url = parse_url($host);
+    $host = isset($url['host'])? $url['host']: $url['path'];
```

Comment:
$url might be false if host is "seriously bad" according to manual. We should handle the case when we can't parse the host.

Reply:
It is best to throw an exception if host can't be parsed or is rather invalid (e.g. empty host/path)...